src/PlusCommon/PlusCommon.h

```diff
@@ -104,6 +104,30 @@ enum PlusImagingMode
   vtkPlusLogger::Instance()->LogMessage(logLevel, msgStream.str().c_str(), __FILE__, __LINE__); \
   }
 
+// If condition is satisfied, logs error only the first time
+// and returns PLUS_FAIL each time
+#define ERROR_FAIL_IF(condition, msg) \

```

Comment:
This is great, but could we add a timeout feature?

In a long run, this error may occur on two separate unique instances, which would be nice to show two error messages.

Maybe 1 message per X minutes?

Reply:
This can be done relatively easily. Will get to it!